src/core/mkContainerFromCabalExe.nix

```diff
@@ -0,0 +1,71 @@
+{ repoRoot, iogx-inputs, user-inputs, pkgs, lib, system, ... }:
+
+userConfig':
+
+let
+  inherit (iogx-inputs.nix2container.packages) nix2container;
+
+  evaluated-modules = lib.evalModules {
+    modules = [{
+      options = lib.iogx.options;
+      config."mkContainerFromCabalExe.<in>" = userConfig';
+    }];
+  };
+
+  userConfig = evaluated-modules.config."mkContainerFromCabalExe.<in>";
+
+  name =
+    if lib.isNull userConfig.name then
+      userConfig.exe.exeName
+    else
+      userConfig.name;
+
+  license =
+    let
+      ls = userConfig.exe.meta.license;
+    in
+    if (lib.isAttrs ls) then
+      ls.spdxId
+    else if (lib.isList ls && [ ] != ls) then
+      let
+        l = (lib.head ls);
+      in
+      if lib.isAttrs l then
+        l.spdxId
+      else
+        null
+    else null;
+
+  labels = lib.pipe
+    {
+      inherit license;
+      inherit (userConfig) description;
+      source = userConfig.sourceUrl;
+    }
+    [
+      (lib.filterAttrs (k: v: v != null))
+      (lib.mapAttrs' (k: v: lib.nameValuePair "org.opencontainers.image.${k}" v))
+    ];
+
+  rootEnv =
+    if (!lib.isNull userConfig.packages && userConfig.packages != [ ]) then
+      pkgs.buildEnv
+        {
+          name = "root";
+          paths = userConfig.packages;
+          pathsToLink = [ "/bin" ];
+        }
+    else
+      null;
+in
+nix2container.buildImage ({
```

Comment:
We should probably have some things here by default (e.g. busybox or some other minimal shell env, cacerts)

Reply:
I was thinking the same thing, but was not sure where to start. I can quickly come up with a small list and link to some scenarios motivating each package.

Reply:
Okay so, as for the minimal shell, why? If its to give the users something to use if they want to run the container in an interactive shell, then we would need to set `Cmd` to the cabal exe instead of `entryPoint` to enable running an interactive shell instead of the exe on invocation of `docker run`. 

The issue with that is that it makes the docker container less usable as a "containerized command" because with cabal exe set to `entryPoint` instead of `Cmd` this allows the user to pass args to the `docker run MY_CONTINER_NAME` command like  `docker run MY_CONTINER_NAME --version` and they go straight to the cabal exe. If we set the exe to `Cmd` passing args becomes much less ergonomic.

Reply:
1818846 Adds `cacret` and `fakeNss`
5e5aaa3 Adds `bash` and `coreutils` and changes from `entryPoint` to `Cmd`. It would have the effect I mentioned if we want to make running an interactive shell in the container easier we can take this commit, and if we want to make passing args to the cabal exe from container invocation easier we can drop it. And if there is a reason to have bash and coreutils outside of interactive shells let me know.

Reply:
We want a shell for something like `kubectl exec`: The container is already running through its proper entrypoint, but I want to connect to the running system and poke around for debugging or emergency maintenance

Reply:
So we can keep the `entryPoint` config and still benefit from a shell in PATH

Reply:
Makes sense, switched back to `entryPoint` included basic `coreutils` and bash.